hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestMaxResultsPerColumnFamily.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.client.TableDescriptor;
+import org.apache.hadoop.hbase.client.TableDescriptorBuilder;
+import org.apache.hadoop.hbase.testclassification.MediumTests;
+import org.apache.hadoop.hbase.testclassification.RestTests;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Category({RestTests.class, MediumTests.class})
+public class TestMaxResultsPerColumnFamily {
+
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestMaxResultsPerColumnFamily.class);
+
+  private static final Logger LOG = LoggerFactory.getLogger(
+    TestMaxResultsPerColumnFamily.class);
+
+  private static final byte [][] FAMILIES = {
+    Bytes.toBytes("1"), Bytes.toBytes("2")
+  };
+
+  private static final byte [][] VALUES = {
+    Bytes.toBytes("testValueOne"), Bytes.toBytes("testValueTwo")
+  };
+
+  private static final HBaseTestingUtil hbaseTestingUtility = new HBaseTestingUtil();
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+    hbaseTestingUtility.startMiniCluster(3);
+  }
+
+  @AfterClass
+  public static void tearDownAfterClass() throws Exception {
+    hbaseTestingUtility.shutdownMiniCluster();
+  }
+
+  @Test
+  public void testSetMaxResultsPerColumnFamilySimple() throws Exception {
+    TableName tableName = TableName.valueOf("TestScannersWithFilters3");
+    Configuration conf = hbaseTestingUtility.getConfiguration();
+    conf.setInt(HConstants.MAJOR_COMPACTION_PERIOD, 0);
+    Admin admin = hbaseTestingUtility.getAdmin();
+    if (!admin.tableExists(tableName)) {
+      ColumnFamilyDescriptorBuilder cfBuilder0 =
+        ColumnFamilyDescriptorBuilder.newBuilder(FAMILIES[0]);
+
+      TableDescriptor tableDescriptor =
+        TableDescriptorBuilder.newBuilder(tableName).setColumnFamily(cfBuilder0.build()).build();
+      admin.createTable(tableDescriptor);
+      Table table = hbaseTestingUtility.getConnection().getTable(tableName);
+
+
+
+      for(int i = 0; i<30000; i++) {
+        byte [] ROW = Bytes.toBytes("" + i);
+        Put p = new Put(ROW);
+
+        p.addColumn(FAMILIES[0], Bytes.toBytes(""+ 1), VALUES[1]);
+        p.addColumn(FAMILIES[0], Bytes.toBytes(""+ 2), VALUES[0]);
+
+        table.put(p);
+      }
+
+      table.close();
+    }
+
+    Table t = hbaseTestingUtility.getConnection().getTable(tableName);
+    int expected = 30000;
+
+    byte[] start = Bytes.toBytes("0");
+    byte[] stop = Bytes.toBytes("F");
+
+    Scan limits = new Scan().withStartRow(start).withStopRow(stop).setReadType(Scan.ReadType.PREAD);
```

Comment:
You're ingesting rowkeys from  `0` to `30000` (in practice 0-9999) so this limit will not reduce the returned rows. Is that intentional?

Reply:
Yes, I want to scan the whole table, i'll remove the start and stop row.